util/src/main/kotlin/DiskCache.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2017 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.util
+
+import ch.frankel.slf4k.error
+
+import com.jakewharton.disklrucache.DiskLruCache
+
+import java.io.File
+import java.io.IOException
+
+/**
+ * Wrapper around [DiskLruCache] that adds a workaround for the 64 character key length limit.
+ */
+class DiskCache(
+        /**
+         * The directory to store the cache data, must be exclusive to the cache.
+         */
+        val directory: File,
+
+        /**
+         * The maximum size of the disk cache.
+         */
+        val maxSize: Long,
+
+        /**
+         * Duration in seconds that cache entries are valid.
+         */
+        val timeToLive: Int
+) {
+    companion object {
+        const val INDEX_FULL_KEY = 0
+        const val INDEX_TIMESTAMP = 1
+        const val INDEX_DATA = 2
+        const val VALUE_COUNT = 3
+    }
+
+    private val diskLruCache = DiskLruCache.open(directory, 0, VALUE_COUNT, maxSize)
+
+    /**
+     * Shorten the string to be usable as a key for [DiskLruCache] which has a maximum length of 64. If the string is
+     * shorter than 57 chars take it as is, otherwise shorten it and append a serial number. The un-shortened string is
+     * stored in the cache to be able to compare if a cache entry matches a key.
+     */
+    private fun String.asKey(): String {
+        return if (length > 57) {
```

Comment:
Extract `57` to a local variable / constant?

Reply:
Done.